operator/cmd/mesh/testdata/manifest-generate/output/all_on.yaml

```diff
@@ -591,14 +591,127 @@ spec:
             mountPath: /etc/prometheus
           - mountPath: /etc/istio-certs
             name: istio-certs
+        - name: istio-proxy
+          image: "gcr.io/istio-testing/proxyv2:latest"
+          ports:
+            - containerPort: 15090
+              protocol: TCP
+              name: http-envoy-prom
+          args:
+            - proxy
+            - sidecar
+            - --domain
+            - $(POD_NAMESPACE).svc.cluster.local
+            - --configPath
+            - "/etc/istio/proxy"
+            - --binaryPath
+            - "/usr/local/bin/envoy"
+            - --serviceCluster
+            - "istio-proxy-prometheus"
+            - --drainDuration
+            - "45s"
+            - --parentShutdownDuration
+            - "1m0s"
+            - --discoveryAddress
+            - istio-pilot.istio-system.svc:15012
```

Comment:
this should not be hardcoded

Reply:
This value is generated automatically by "go test" based on the configuration in manifests/istio-telemetry/prometheus/templates/deployment.yaml, which is configured to be the same as --discoveryAddress in
manifests/gateways/istio-ingress/templates/deployment.yaml.

In manifests/istio-control/istio-discovery/files/injection-template.yaml, there is a different way of configuring discoveryAddress:
    - --discoveryAddress
    - "{{ annotation .ObjectMeta `sidecar.istio.io/discoveryAddress` .ProxyConfig.DiscoveryAddress }}"

But the "annotation" directive is not understoodable by the processor of manifests/istio-telemetry/prometheus/templates/deployment.yaml.